### DISCUSSION TITLE
Add shadows to the desktop icon labels for better readability

### DISCUSSION THREAD

**Discussion by moloch1994** [linuxmint/discussions#296] [Ideas]
Hi.

If you look at Windows or many other desktop environments like KDE or Xfce, they tend to have shadows behind the desktop icon labels (the text under the icons). Cinnamon lacks such feature. While it looks okay with dark wallpapers, as soon as you use a bright wallpaper, it becomes unreadable. For demonstration I attach screenshots without and with shadows:

![Screenshot from 2020-01-21 20-41-11](https://github.com/linuxmint/discussions/assets/68343381/1495ea0e-b251-4017-9afe-7f07aa32dc40)
![Screenshot from 2024-01-19 20-31-56](https://github.com/linuxmint/discussions/assets/68343381/14eef185-2555-4c78-9b0b-0383df009959)

The first screenshot is from ZorinOS, but nevermind that (I just already have shadows enabled on my system, so I don't have any Mint screenshots without them), the text labels look the same on Mint anyway. I think either way you can see how much shadows improve readability, which is why they've been standard on Windows since early 2000s (and before that they used colored backgrounds for labels, again for readability).

I don't see any downside to implementing shadows. People who don't need them because they use dark backgrounds wouldn't even notice them. I also don't see why this couldn't be implemented immediately. After all, the solution already exists, and it's dead simple.

Create a file called gtk.css in /home/user/.config/gtk-3.0 with the following contents:

`.nemo-desktop.nemo-canvas-item {
    color: #fff;
    text-shadow: 1px 1px 2px #000, 1px 1px #000, 1px 1px #000, 1px 1px #000,  2px 2px 3px #000;
}
.nemo-desktop.nemo-canvas-item:hover {
    background-color: alpha(@theme_selected_bg_color, 0.33);
    background-image: none;
}
.nemo-desktop.nemo-canvas-item:selected {
    background-color: u/theme_selected_bg_color;
    background-image: none;
    color: u/theme_selected_fg_color;
    text-shadow: none;
}`

The numerical values can be tweaked to increase or decrease the effect. Also, the same effect can be achieved on Mate, if you substitute every mention of nemo in the code for caja.

Every time I install Linux Mint, I end up just copy-pasting this file manually. I don't see why it couldn't just be shipped with the distro. I don't see how it's anything but an improvement. And the change would literally take a minute to implement.

I hope Mint developers will see this message. If you disagree with me, please explain why.

**Comment by 100savage** [comment]
I like this. Is there a way to add shadows to the file name in Nemo?

**Comment by 100savage** [comment]
I started to play with the gtk.css file that you created. I was able to make the shadows globally.

```
* {
    color: #fff;
    text-shadow: 4px 4px 4px #000;
}
```
Have fun and tweak it as needed

**Comment by anaximeno** [comment]
Hi @moloch1994, I believe the repo https://github.com/linuxmint/mint-themes would be the right place for someone to create a pull request suggesting this change. 

#### Edit
I've already created a pull request at the repository for this https://github.com/linuxmint/mint-themes/pull/466

**Comment by Secret-chest** [comment]
Yes, it's a theme thing.

**Comment by Secret-chest** [comment]
![image](https://github.com/linuxmint/discussions/assets/74449186/5f17753d-4be0-47ef-9768-548bbc779b02)
With Materia you get shadows.

**Comment by mfreeman72** [comment]
@moloch1994 - I've been adding this exact same gtk.css code to every install of Linux Mint for years. I've also made this exact same feature request before. Sadly, there wasn't even a single comment made on the thread. It baffles me why this wouldn't just be something they'd do by default, since the usual hard-line, one-pixel shadow looks so incredibly unreadable on lighter backgrounds. I also use a bold font for icon labels. That makes them look much more readable, in addition to better shadows. Since no one seems to want this but you and me, apparently, I just use a separate /home partition, so I can just keep little niceties like this across upgrades and reinstalls and never have to worry about some of the more rough parts of Linux Mint (I've also tweaked their Mint-X theme to be a bit nicer looking -- with slightly darker and inset scroll bars, rounded menu corners, etc.).

@Secret-chest @anaximeno - It's not a theme thing. Personally, I wouldn't want a theme change to mess with desktop icon label shadows at all. It's the one appearance thing that makes absolutely no sense to have theme-changable. Maybe add it as a desktop configuration option instead, if you want it to be changable. Putting it in individual themes sounds like a horrible way to do things, since I would want this to be in place, no matter what theme I might want to install.

**Reply by Secret-chest** [reply]
Then use gtk.css to override the theme.

**Reply by mfreeman72** [reply]
Yes, as I just said, that's what I already do. I just believe there should be better defaults than what there is, as I keep having to change it every time I install Linux Mint on a new machine, which is frustrating and bad. And I think that putting it in themes makes no sense whatsoever. The default appearance is incredibly ugly.

**Reply by moloch1994** [reply]
I agree, this is one of those things that I'm kinda baffled that most people don't even notice it or pay attention. I've been a Linux user for 4 years now, and this was something I felt the need to address within like the first 3 months because I couldn't use any bright wallpapers. It's not difficult for me to copy-paste the gtk.css file (I keep it in a cloud), but it's always been something that makes it harder to introduce other people into Linux, because Mint is the distro I always recommend, but then I have to explain to them this aspect. I'm glad they're adding the shadows to the main theme at least. But I agree with you, I'd rather it was a universal feature.

**Reply by mfreeman72** [reply]
***EDIT: Please skip to the next comment for a newer version of this program.***

Well, I put together a quick and dirty Python-based GUI for setting and adjusting this. For now, this could be a quick-fix to prevent having to hand-code this every time. It basically lets you turn on/off the drop shadow override in gtk.css (so you can select either what's built into the theme you're using, or override those to the values you want), and adjust the shadow gradient radius, the vertical distance from center, and horizontal distance from center. I'll refine this as I have time, but this is the kind of thing I'd love to see put into the System Settings > Desktop dialog. Since I can't upload a python file here, I'll share a Google Drive link...

https://drive.google.com/file/d/1T21NxSmSVu1R-z1a129w3ts4huaXcAYe/view?usp=sharing

Start by taking gtk.css back to its default, or you may run into trouble. After making any changes, log out/in or reboot to see the changes made.

*BE SURE TO MAKE A BACKUP COPY OF YOUR GTK.CSS FILE BEFORE RUNNING THIS!!!*

**Reply by mfreeman72** [reply]
***Updated version.*** I adjusted the window appearance, and added a BASH script to run it from, which reloads Nemo when you close the window. I've left the old version at the previous link if you want to see the difference.

Download this file: https://drive.google.com/file/d/1LYgZI6EoPZCpQX2l1SDKYifcMGQlPjjC/view?usp=drive_link

Make sure to start with a clean gtk.css file, taking it back to defaults, or just renaming it to keep a backup of your previous edits (the program will create a new one). The program does work around any other css code that's there, but if you have two of the same css elements in the same file, I can't guarantee it will work the way it's supposed to.

Decompress the folder wherever you want to store it, and run the .sh file. The window will open. Click the Enable switch, adjust the settings the way you want them, and when you close the window, Nemo will restart and your new shadow settings will be visible.
